TESTING.md

```diff
@@ -1,70 +1,88 @@
-    * Server: 172.17.42.1
-    * Port: 3128
-5. Optionally, you can click `Advanced`, and then test `http://www.google.com` and you should see the output in the tcpdump terminal window.
-6. Finally, run the test build again.
-7. You should now see output from the tcpdump terminal window!
-
-Repeat this by enabling/disabling the proxy configuration in Jenkins and repeating the test to prove to yourself that it was not a fluke.
-**Note: I've noticed that Jenkins seems to send a disconnection series of packets to the proxy server with the first connection attempt (when you run your test build), after you've disabled the proxy configration. Run the test job a second time, and you'll again not see any output.**
+    ```
+    version: "3.7"
+    services:
+      jenkins:
+        image: jenkins/jenkins:lts
+        ports:
+          - 8080:8080
+        volumes:
+          - $JENKINS_PLUGIN/target/:/var/jenkins_home/plugins
+      
+      dogstatsd:    
+        image: datadog/dogstatsd:latest 
+        environment:
+          - DD_API_KEY=<API_KEY>
+          - DD_LOG_LEVEL=debug
+        ports:
+          - 8125:8125/udp  
+               
+    ```
 
-## Check Style
-In order to check that the Java style meets the recommendations of Sun and Google, as closely as possible, here is a way to test it.
+2. Set the `JENKINS_PLUGIN` environment variable to point to the folder of the clone/fork of this repository.
+3. Run `docker-compose -f <DOCKER_COMPOSE_FILE_PATH> up`.
+4. Check your terminal and look for the admin password:
+    ```
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    jenkins_1    |
+    jenkins_1    | Jenkins initial setup is required. An admin user has been created and a password generated.
+    jenkins_1    | Please use the following password to proceed to installation:
+    jenkins_1    |
+    jenkins_1    | 70dfe3d2b23841cc96c101c0521716f0
+    jenkins_1    |
+    jenkins_1    | This may also be found at: /var/jenkins_home/secrets/initialAdminPassword
+    jenkins_1    |
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    jenkins_1    | *************************************************************
+    ``` 
+
+5. Access your Jenkins instance http://localhost:8080
+6. Enter the administrator password in the Getting Started form.
+7. On the next page, click on the "Select plugins to be installed" unless you want to installing suggested plugins. 
+8. Selected desired plugins depending on your needs. You can always add plugins later.
+  - You can skip user creation and stick to the admin user.
+9. Create a user so that you don't have to use the admin credential again (optional)
```

Comment:
This and the note under `8` seem to contradict a little. Could we remove the last piece since we state this is `optional` anyway?